tsconfig.json

```diff
@@ -0,0 +1,37 @@
+{
+  "compilerOptions": {
+    "target": "ES2018",
+    "module": "ESNext",
+    "moduleResolution": "Node",
+    "lib": [
+      "ESNext",
+      "ESNext.AsyncIterable",
+      "DOM"
+    ],
+    "esModuleInterop": true,
+    "allowJs": true,
+    "sourceMap": true,
+    "strict": true,
+    "noEmit": true,
+    "experimentalDecorators": true,
```

Comment:
can we add more rules? Better do it now than when it's too late:

- noImplicitAny: drastic but efficient to properly type functions
- noUncheckedIndexedAccess : makes sure all properties of an object are typed
- noUnusedLocals: remove imports that are unused
- noUnusedParameters: removes functions params that are unused
- forceConsistentCasingInFileNames: self-explanatory. Allows to detect problems coming from windows user for linux users